### PR TITLE
DataLinks: Ensure window origin to be stripped from url when using absolute urls

### LINKED ISSUE #32555: DataLinks: DataLinks with absolute links are not working
Tested with stat panel with a absolute url to another dashboard 

Cause: React router change

### PR DESCRIPTION
**Pull request by dprokop**
When the appSubUrl was not configured and absolute path was provided, `stripBaseFromUrl` function  returned original passed **absolute URL**.

Fixes https://github.com/grafana/grafana/issues/32555

### CODE REVIEW

File: public/app/core/navigation/patch/interceptLinkClicks.ts [hunk 1]
```diff
@@ -21,9 +21,17 @@ export function interceptLinkClicks(e: MouseEvent) {
       // Ensure old angular urls with no starting '/' are handled the same as before
```
Comment (dprokop) [inline public/app/core/navigation/patch/interceptLinkClicks.ts:21]:
```suggestion
      // Ensure old angular urls with no starting '/' are handled the same as before
      // Make sure external links are handled correctly
```

File: packages/grafana-data/src/utils/location.test.ts [hunk 2]
```diff
@@ -1,18 +1,65 @@
...
+        getTimeRangeForUrl: (() => {}) as any,
+      });
+    });
+
+    test('relative url', () => {
+      const urlWithoutMaster = locationUtil.stripBaseFromUrl('/subUrl/grafana/');
+      expect(urlWithoutMaster).toBe('/subUrl/grafana/');
+    });
+
+    test('absolute url', () => {
```
Comment (torkelo) [inline packages/grafana-data/src/utils/location.test.ts:60]: This scenario seem incorrect to me, an absolute url should stay absolute

File: packages/grafana-data/src/utils/location.ts [hunk 3]
```diff
@@ -14,10 +14,16 @@ let getVariablesUrlParams: (scopedVars?: ScopedVars) => UrlQueryMap;
 const stripBaseFromUrl = (url: string): string => {
   const appSubUrl = grafanaConfig.appSubUrl ?? '';
   const stripExtraChars = appSubUrl.endsWith('/') ? 1 : 0;
-  const urlWithoutBase =
-    url.length > 0 && url.indexOf(appSubUrl) === 0 ? url.slice(appSubUrl.length - stripExtraChars) : url;
+  const isAbsoluteUrl = !url.startsWith('/');
```
Comment (torkelo) [inline packages/grafana-data/src/utils/location.ts:17]: If it starts with http its also absolute

**Review by torkelo — COMMENTED** [review]
A bit confused by this, seem to change stuff that is not about external/full http urls but absolute to root url / as well. Is this to avoid full page reloads when going to http full url that is to the same Grafana instance?

File: packages/grafana-data/src/utils/location.ts (hunk 3 above)
Comment (dprokop) [reply packages/grafana-data/src/utils/location.ts:17]: it covers it, doesn't it?

**Comment by torkelo** [comment]
> When the appSUbUrl was not configured, stripBaseFromUrl function returned original passed URL.


Is that not correct?

**Comment by dprokop** [comment]
> Is this to avoid full page reloads when going to http full url that is to the same Grafana instance?

yes

**Comment by dprokop** [comment]
@torkelo : updated description, I wasn't precise:

File: packages/grafana-data/src/utils/location.test.ts (hunk 2 above)
Comment (dprokop) [reply packages/grafana-data/src/utils/location.test.ts:60]: Well, yes and no. This function is used for getting everything after the appSubUrl segment in the URL. I think it shouldn't care whether or not the input URL is absolute or relative. Given `[url|/url]/appSubPath/pathname` should just give pathname. Or the input if there is nothing to strip.

File: packages/grafana-data/src/utils/location.ts (hunk 3 above)
Comment (torkelo) [reply packages/grafana-data/src/utils/location.ts:17]: but should it not be: 

const isAbsolute = url.startsWith(`http`)   

I am not sure I understand the function, an url like `datasources/new`  does not starts with `/` , so isAbsolute would be true

File: packages/grafana-data/src/utils/location.test.ts [hunk 4]
```diff
@@ -1,18 +1,65 @@
...
+      locationUtil.initialize({
+        config: {} as any,
+        getVariablesUrlParams: (() => {}) as any,
+        getTimeRangeForUrl: (() => {}) as any,
+      });
+    });
+
+    test('relative url', () => {
+      const urlWithoutMaster = locationUtil.stripBaseFromUrl('/subUrl/grafana/');
+      expect(urlWithoutMaster).toBe('/subUrl/grafana/');
```
Comment (torkelo) [inline packages/grafana-data/src/utils/location.test.ts:57]: I would label this absolute also (as it starts with `/`)

**Review by torkelo — APPROVED** [review]
I understand the need now for replacing fully qualified urls, but some of logic is a bit unclear I think. (isAbsolute)

File: packages/grafana-data/src/utils/location.test.ts (hunk 4 above)
Comment (dprokop) [reply packages/grafana-data/src/utils/location.test.ts:57]: Absolute URLs does not start with `/`, are you sure about this comment @torkelo?

Comment (torkelo) [reply packages/grafana-data/src/utils/location.test.ts:57]: If it starts with / it is absolute to root / base href

**Pull request merged**